graphql/executor.go

```diff
@@ -229,6 +229,7 @@ func SafeExecuteResolver(ctx context.Context, field *Field, source, args interfa
 
 type ExecutorRunner interface {
 	Execute(ctx context.Context, typ Type, source interface{}, query *Query) (interface{}, error)
+	ExecuteWithPartialFailures(ctx context.Context, typ Type, source interface{}, query *Query) (interface{}, []error, error)
```

Comment:
I wanted it to be a separate function to keep it backward compatible, but Im not sure I like the naming/pattern, open to suggestions